spec/services/submission/trait_score_parser_spec.rb

```diff
@@ -108,13 +241,87 @@
     end
 
     it 'handles empty newlines properly' do
-      input_is "plant 1,1  \n\nplant X,\nplant 2, 2\n\n"
+      input_is "plant 1,pa,oo,pl,1  \n\nplant X,pa,oo,pl,\nplant 2,pa,oo,pl, 2\n\n"
       subject.send(:parse_scores)
       expect(subject.trait_scores).
         to eq({ 'plant 1' => {0 => '1'},
                 'plant 2' => {0 => '2'},
                 'plant X' => {} })
     end
+
+    it 'requires each line to provide accession-related information' do
+      input_is "plant 1,pa,oo,pl\nplant X,pa\nplant 2\n"
+      subject.send(:parse_scores)
+      expect(subject.trait_scores).
+        to eq({ 'plant 1' => {} })
+      expect(upload.logs).
+        to include 'Ignored row for plant X since either Plant accession or Originating organisation is missing.'
+      expect(upload.logs).
+        to include 'Ignored row for plant 2 since either Plant accession or Originating organisation is missing.'
+    end
+
+    it 'parses accession information to a dedicated structure' do
+      input_is "plant 1,pa,oo,pl\nplant X,pa\nplant Y\nplant 2,pa2,oo,pl"
+      subject.send(:parse_scores)
+      expect(subject.accessions).
+        to eq({ 'plant 1' => { plant_accession: 'pa', originating_organisation: 'oo' },
+                'plant 2' => { plant_accession: 'pa2', originating_organisation: 'oo' }})
+    end
+
+    it 'ignores scores beyond number of trait columns' do
```

Comment:
Shouldn't that also give a warning? Clearly, something is wrong with the uploaded data.